aten/src/ATen/native/NativeFunctions.cpp

```diff
@@ -595,5 +597,75 @@ Tensor RoiPooling2d_backward_cpu(
   throw std::runtime_error("not implemented");
 }
 
+
+
+// TODO Replace this with more accurate digamma().
+template <typename CScalar>
+static inline CScalar digamma_one(CScalar x) {
+  const CScalar eps = x * 1e-2;
+  return (std::lgamma(x + eps) - std::lgamma(x - eps)) / (eps + eps);
+}
+
+/** Computes the reparameterized gradient -(d/dalpha cdf(x;alpha)) / pdf(x;alpha)
+    for random number x drawn from a standard Gamma distribution Gamma(alpha).
+*/
+template <typename CScalar>
```

Comment:
I know this is bikeshedding, but the `CScalar` name makes me cringe. Even `scalar` or `number` or `real` would make me happier. (Lower-case makes me think it corresponds to a built-in type and not an aggregate/struct type)

Reply:
ok, going with `scalar` in a  future commit unless there are objections.